website/src/css/custom.css

```diff
@@ -37,6 +37,13 @@
   --docusaurus-highlighted-code-line-bg: rgba(0, 0, 0, 0.3);
 }
 
+/* Make hero banner less ugly for dark theme */
+ [data-theme='dark'] .hero--primary {
+  /** Defaults to primary color which is horrible for the hero background */
+  --ifm-hero-background-color: #2b307d; /* --ifm-color-primary-lightest from light theme */
+  --ifm-hero-text-color: var(--ifm-font-color-base);
```

Comment:
I think you should add these to the `[data-theme='dark']` rule above (lines 29–38), since it looks the variable name itself has "hero" in its name already, so there's no need to specifically select the `.hero--primary` element. This would allow the variables to be accessible for all elements under element that has the `theme="dark"` attribute set.

To help explain this a bit more, the formal name for CSS variables is [custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties), and they actually behave exactly normal properties (e.g. `font-size`, etc.) except that they have no builtin behavior. They are "cascaded" ("inherited") by all elements below them, so if you treat them as "variables", then it's as if the variable is scoped to all descendant elements. Also, in case if you're not familiar, `[data-theme='dark']` is the syntax for selecting an element based on its attributes, and the `data-` prefix is specifically for custom attributes, so it implies that there is an actual element in the DOM somewhere that has `data-theme="dark"` set somewhere, likely somewhere very high in the hierarchy. The values of the variables can change dynamically, if the `data-theme` attribute's value is changed dynamically, causing the other selectors to be applied and cascade down the tree.

The tl;dr for the previous paragraph is that you should basically treat the first two blocks in this file as essentially defining "global" variables that are conditionally applied based on the value of the `data-theme` attribute, and you should probably declare all other variables in those blocks unless you really want to override the variable for something more specific. I've been saying "variable" but they are really "constants" for all intents and purposes, since CSS rules cannot change the value of a variable as seen by a higher scope.

For consistency, you might also consider putting the light mode version in the `:root` rule above it, rather than relying on the default.

Reply:
I actually tried that originally, and as far as I can tell, it doesn't work, maybe we should chat offline about why because I have no idea 😂